src/Stratis.Bitcoin/Consensus/PeerGetHeaderRequestBehavior.cs

```diff
@@ -0,0 +1,132 @@
+    /// within a certain time frame.
+    /// </summary>
+    /// <remarks>
+    /// This behavior is a tempory work around to peers that spam the node with too many <see cref="GetHeadersPayload"/> requests.
+    /// It will be changed in the future once a more in-depth and thorough implementation has been agreed upon.
+    /// </remarks>
+    public sealed class RateLimitingBehavior : NetworkPeerBehavior
+    {
+        /// <summary>How long the offending peer will be banned for.</summary>
+        private const int BanDurationSeconds = 3600;
+
+        /// <summary>Provider of time functions.</summary>
+        private readonly IDateTimeProvider dateTimeProvider;
+
+        /// <summary>The last header that was requested.</summary>
+        private uint256 getHeaderLastRequestHash;
+
+        /// <summary>The last time the header was requested.</summary>
+        private DateTimeOffset? getHeaderLastRequestedTimestamp;
+
+        /// <summary>The amount of times the same request has been made.</summary>
+        private int getHeaderRequestCount;
+
+        /// <summary>The threshold after which the node will be banned and disconnected.</summary>
+        private const int GetHeaderRequestCountThreshold = 10;
+
+        /// <summary>Instance logger.</summary>
+        private readonly ILogger logger;
+
+        /// <summary>Factory for creating loggers.</summary>
+        private readonly ILoggerFactory loggerFactory;
+
+        /// <summary>Contract for peer banning behavior.</summary>
+        private readonly IPeerBanning peerBanning;
+
+        public RateLimitingBehavior(
+            IDateTimeProvider dateTimeProvider,
+            ILoggerFactory loggerFactory,
+            IPeerBanning peerBanning)
+        {
+            this.dateTimeProvider = dateTimeProvider;
+            this.loggerFactory = loggerFactory;
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
+            this.peerBanning = peerBanning;
+        }
+
+        public override object Clone()
+        {
+            return new RateLimitingBehavior(this.dateTimeProvider, this.loggerFactory, this.peerBanning);
+        }
+
+        protected override void AttachCore()
+        {
+            this.AttachedPeer.MessageReceived.Register(this.OnMessageReceived, true);
+        }
+
+        protected override void DetachCore()
+        {
+            this.AttachedPeer.MessageReceived.Unregister(this.OnMessageReceived);
+        }
+
+        /// <summary>
+        /// Processes an incoming message from the peer.
+        /// </summary>
+        /// <param name="peer">Peer from which the message was received.</param>
+        /// <param name="message">Received message to process.</param>
+        private Task OnMessageReceived(INetworkPeer peer, IncomingMessage message)
+        {
+            switch (message.Message.Payload)
```

Comment:
no need for switch. Just check if payload is getheaders and if it is call the handler method

Reply:
@dangershony  asked that we keep this switch if we perhaps want to handle other payload types?

Reply:
We are not going to handle other payload types in the future because this PR is a fix for a bug in our nodes. Preparing for having similar bug but that spams other payloads is not necessary imho. 

Throttling solution will have a different design and will not be done on top of this PR

Reply:
Ok yeah I dont mind either way, will change 👍

Reply:
I disagree, first we use switch every where in other places why not here (consistency) 
Second later we will reuse this behaviour for rate limiting (as agreed already)

Reply:
I remember this point pops out every time with the same conclusion :)
(I'm fine with consistency)